crates/neon-runtime/src/napi/bindings/mod.rs

```diff
@@ -0,0 +1,80 @@
+macro_rules! napi_name {
```

Comment:
Sample macro output:

```rust
#[cfg(not(windows))]
use libloading::os::unix::Library;
use std::os::raw::{c_char, c_void};
use super::types::*;

pub(crate) struct Napi {
    get_undefined: unsafe extern "C" fn(env: Env, result: *mut Value) -> Status,
    /* ... repeat for each N-API function */
}

pub(crate) unsafe fn load() -> Result<(), libloading::Error> {
    let host = Library::this();

    NAPI = Napi {
        get_undefined: *host.get("napi_get_undefined".as_bytes())?,
        /* ... repeat for each N-API function */
    };

    Ok(())
}

#[inline]
pub(crate) unsafe fn get_undefined(env: Env, result: *mut Value) -> Status {
    (NAPI.get_undefined)(env, result)
}

fn panic_load<T>() -> T {
    panic!("Must load N-API bindings")
}

static mut NAPI: Napi = {
    unsafe extern "C" fn get_undefined(_: Env, _: *mut Value) -> Status {
        panic_load()
    }
    /* ... repeat for each N-API function */

    Napi {
        get_undefined,
        /* ... repeat for each N-API function */
    }
};
```

Reply:
Need to add this as a comment to more clearly document the output of the macro.

Reply:
This example is so helpful. Can you put it into a comment directly in the source file, maybe with a few sentences explaining the overall approach?